net/src/ipv6/mod.rs

```diff
@@ -29,10 +29,20 @@ impl Ipv6 {
     #[allow(unsafe_code)]
     pub const MIN_LEN: NonZero<usize> = unsafe { NonZero::new_unchecked(40) };
 
+    /// Create a new [`Ipv6`] header
+    ///
+    /// # Errors
+    ///
+    /// Returns an [`Ipv6Error::InvalidSourceAddr`] error if the source address is invalid.
+    pub fn new(header: Ipv6Header) -> Result<Self, Ipv6Error> {
```

Comment:
Not strictly required for this PR, but on a longer timeline it would be ideal to remove `etherparse` types from the public api of this crate (we can't 1.0 this crate until that happens).  That said, this is hardly the only place in the crate where we expose those types so :shrug:

Reply:
Yep, we can address this issue in bulk later.